### PR TITLE
test: updated convergence catx and contest

### PR DESCRIPTION
**Pull request by WissBe**

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -12,8 +12,8 @@ For simplicity, we normalize the feature and action spaces.
 ```python
 from typing import Optional
 import tensorflow as tf
-import numpy as np
 from sklearn.datasets import fetch_openml
+import numpy as np
 from catx.type_defs import Actions, Costs, Observations
```
Comment (cyprienc) [inline docs/getting_started.md:17]: does this actually pass linters?

File: docs/getting_started.md [hunk 2]
```diff
@@ -94,22 +111,43 @@ def main() -> None:
...
             break
-        actions, probabilities = catx.sample(obs=obs, epsilon=0.05)
+
+        if i == 0:
+            network_extras = {"dropout_rate": 0.2}
+            state = catx.init(
+                obs=obs, epsilon=epsilon, key=key, network_extras=network_extras
+            )
+
+        state.network_extras["dropout_rate"] = 0.2
```
Comment (cyprienc) [inline docs/getting_started.md]: Nitpicking here but maybe dropout should be 0 for sampling and 0.2 for learning no?

File: docs/getting_started.md [hunk 3]
```diff
@@ -94,22 +111,43 @@ def main() -> None:
...
+        actions, probabilities, state = catx.sample(
+            obs=obs, epsilon=epsilon, state=state
+        )
+        actions = np.array(actions)
+        probabilities = np.array(probabilities)
```
Comment (cyprienc) [inline docs/getting_started.md]: are these two lines necessary?

File: tests/conftest.py [hunk 4]
```diff
@@ -1,25 +1,70 @@
...
+        obs: Observations,
+        key: PRNGKey,
+        network_extras: NetworkExtras,
+    ) -> Logits:
+        return self.network(obs)
+
+
+# class MLPBuilder(NetworkBuilder):
+#     def create_network(self, depth: int) -> hk.Module:
+#         return hk.nets.MLP([3] + [2 ** (depth + 1)], name=f"mlp_depth_{depth}")
```
Comment (cyprienc) [inline tests/conftest.py]: commented code

File: tests/conftest.py [hunk 5]
```diff
@@ -67,3 +112,10 @@ def costs() -> ArrayNumpy:
 @pytest.fixture
 def epsilon() -> float:
     return 0.0
+
+
+@pytest.fixture
+def key() -> PRNGKey:
+    key = jax.random.PRNGKey(42)
+    key, subkey = jax.random.split(key, num=2)
```
Comment (cyprienc) [inline tests/conftest.py]: is this split necessary?

File: tests/test_catx.py [hunk 6]
```diff
@@ -1,38 +1,96 @@
...
+        optimizer=optax.adam(learning_rate=0.01),
+        discretization_parameter=4,
+        bandwidth=1.5 / 4,
+        action_min=action_min,
+        action_max=action_max,
+    )
+
+    state = catx.init(obs=jax_observations, epsilon=epsilon, key=key)
+
+    return catx, state
```
Comment (cyprienc) [inline tests/test_catx.py]: these two fixtures are quite similar. Could `catx_init` actually use the fixture `catx`

File: tests/test_catx.py [hunk 7]
```diff
@@ -42,6 +100,38 @@ def catx(mlp_builder: NetworkBuilder, request: pytest.FixtureRequest = None) ->
...
+        action_min=action_min,
+        action_max=action_max,
+    )
+
+    network_extras = {"dropout_rate": 0.2}
+    state = catx.init(
+        obs=jax_observations, epsilon=epsilon, key=key, network_extras=network_extras
+    )
+
+    return catx, state
```
Comment (cyprienc) [inline tests/test_catx.py]: it seems to me that we are creating a lot of fixtures that are only used once. 
Could a function do the trick and replace all these fixtures above?
